### ISSUE TITLE
functions that are used in functional style rather than standard don't resolve symbols

### ISSUE THREAD

**Issue by SingingBush** [intellij-dlanguage/intellij-dlanguage#1014]
functions that are used in functional style rather than standard don't resolve symbols. For example, see below usage of `std.stdio : writeln`:

![image](https://github.com/user-attachments/assets/b8050fb6-40e1-4425-899f-2e2bd1afb2cf)

**Comment by rikkimax** [comment]
That would be UFCS not working.

**Comment by etienne02** [comment]
> That would be UFCS not working.

Exactly, I didn’t added the UFCS support for now when I re-implemented the resolve. My fear was to add something and to need to redo it when I add type support (member search basically). Now I understand better the algorithm, I know that it’s possible to properly re-add UFCS without conflicting with the future type implementation.
Basically in the resolveAndWalk you need to search for top level declarations (including imported) if the previous search did not succeed.
